src/scenes/ObservationsRouter/common/MonitoringPlotEditPhotos.tsx

```diff
@@ -0,0 +1,187 @@
+
+  const { adHocObservationResults, observationResults } = usePlantingSiteData();
+
+  const visibleMediaItems = useMemo(
+    () => mediaItems.filter((item) => item.type === 'new' || !item.isDeleted),
+    [mediaItems]
+  );
+
+  const onSetFiles = useCallback((files: File[]) => {
+    const newPhotos: MonitoringPlotMediaItem[] = files.map((file) => ({
+      data: { file, isOriginal: false, type: 'Plot', mediaKind: isVideoFile(file) ? 'Video' : 'Photo' },
+      type: 'new' as const,
+    }));
+    setMediaItems((prevPhotos) => [...prevPhotos, ...newPhotos]);
+  }, []);
+
+  const getUpdatePhotoCaption = useCallback(
+    (index: number) => (caption: string) => {
+      const updatedPhotos = mediaItems.map((mediaItem, i) => {
+        if (index !== i) {
+          return mediaItem;
+        }
+
+        if (mediaItem.type === 'new') {
+          return {
+            ...mediaItem,
+            data: { ...mediaItem.data, caption },
+          };
+        } else {
+          return {
+            ...mediaItem,
+            data: { ...mediaItem.data, caption },
+            isModified: true,
+          };
+        }
+      });
+      setMediaItems(updatedPhotos);
+    },
+    [mediaItems]
+  );
+
+  const getDeletePhoto = useCallback(
+    (index: number) => () => {
+      const mediaItem = mediaItems[index];
+      if (mediaItem.type === 'new') {
+        // remove new photos completely
+        const updatedPhotos = mediaItems.filter((_, i) => index !== i);
+        setMediaItems(updatedPhotos);
+      } else {
+        // mark existing photos as deleted
+        const updatedPhotos = mediaItems.map((item, i) => {
+          if (index === i) {
+            return {
+              ...item,
+              isDeleted: true,
+            };
+          }
+          return item;
+        });
+        setMediaItems(updatedPhotos);
+      }
+    },
+    [mediaItems]
+  );
+
+  const result = useMemo(() => {
+    if (!Number.isNaN(observationId)) {
+      return (
+        observationResults?.find((_result) => _result.observationId === observationId) ??
+        adHocObservationResults?.find((_result) => _result.observationId === observationId)
+      );
+    }
+  }, [adHocObservationResults, observationId, observationResults]);
+
+  useEffect(() => {
+    if (result) {
+      result.plantingZones.forEach((zone) =>
+        zone.plantingSubzones.forEach((subzone) =>
+          subzone.monitoringPlots.forEach((plot) => {
+            if (plot.monitoringPlotId === monitoringPlotId) {
+              setMonitoringPlotResult(plot);
+              setMediaItems(plot.media.map((mediaElement) => ({ type: 'existing', data: mediaElement })));
+              return;
+            }
+          })
+        )
+      );
+    }
+  }, [result, monitoringPlotId]);
+
+  const goToPhotosTab = useCallback(() => {
+    navigate(APP_PATHS.OBSERVATIONS);
+  }, [navigate]);
+
+  const saveActivity = useCallback(() => {
+    // save media items to backend
```

Comment:
Same question here @tommylau523 . Would it be better if I add the "redux" stuff for saving the files with RTK query or just normal redux?